v-next/hardhat-node-test-coverage/src/collector.ts

```diff
@@ -0,0 +1,44 @@
+import type { CoverageReport } from "hardhat/types/coverage";
```

Comment:
I was expecting this file to live in the coverage plugin, so that we don't create a new package/dependency.

Reply:
Actually, we probably want it in the `node-test-runner` plugin instead.

Reply:
Got you! My thinking was that it would be good to keep it separate but seeing how tiny it's going to be in the end, I don't think there's much point to it.

Reply:
I deleted the `hardhat-node-test-coverage` package here: https://github.com/NomicFoundation/hardhat/pull/6595/commits/5f11babac5f45b0eeaa381420d513fb1e4f33b76

And replaced it with a file within `node-test-runner` here: https://github.com/NomicFoundation/hardhat/pull/6595/commits/6214a39955f6ff99c1c996c015b096759a011370